tests/nested/manual/core20-install-device-file-install-via-hook-hack/task.yaml

```diff
@@ -0,0 +1,159 @@
+    mkdir -p pc-gadget/meta/hooks
+    cp install-device pc-gadget/meta/hooks/install-device
+    cp prepare-device pc-gadget/meta/hooks/prepare-device
+
+    echo "Add the extra hooks definition to the snap.yaml"
+    # convert our yaml files to json (yaml is easier to write and maintain in VCS)
+    yaml2json < pc-gadget/meta/snap.yaml | tee pc-gadget/meta/snap.json > /dev/null
+    yaml2json < snap-yaml-extras.yaml | tee snap-yaml-extras.json > /dev/null
+    # slurp our two json files together into one json document, then convert 
+    # back to yaml and write out to the snap.yaml in the unpacked gadget snap
+    jq -s '.[0] * .[1]' <(cat snap-yaml-extras.json) <(cat pc-gadget/meta/snap.json) | json2yaml | tee pc-gadget/meta/snap.yaml
+
+    # delay all refreshes for a week from now, as otherwise refreshes for our 
+    # snaps (which are asserted by the testrootorg authority-id) may happen, which
+    # will break things because the signing keys won't match, etc. and 
+    # specifically snap-bootstrap in the kernel snap from the store won't trust
+    # the seed keys to unlock the encrypted data partition in the initramfs
+    sed defaults.yaml -e "s/HOLD-TIME/$(date --date="next week" +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+      pc-gadget/meta/gadget.yaml
+
+    snap pack pc-gadget/ extra-snaps/
+
+    # start fake device svc
+    systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
+    tests.cleanup defer systemctl stop fakedevicesvc
+
+    # include the extra snap declaration bits we need to auto-connect the 
+    # system-files plugs for the install-device hook
+    # XXX: this is a bit of a hack, we want the snap to be able to use the
+    # system-files interface from the install-device hook, so here we just allow
+    # any system-files interface plug to be used with the snap, in reality this
+    # rule would be much more specific
+
+    NESTED_FAKESTORE_SNAP_DECL_PC_GADGET="pc-snap-decl-extras.json"
+    export NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+    "$TESTSTOOLS"/nested-state build-image core
+    unset NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+
+    "$TESTSTOOLS"/nested-state create-vm core
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    echo "Check we have the right model from snap model"
+    nested_exec exec "sudo snap model --verbose" | MATCH "model:\s+testkeys-snapd-secured-core-20-amd64"
+    nested_exec exec "sudo snap model --verbose" | MATCH "grade:\s+secured"
+    nested_exec exec "sudo snap model --verbose --serial" | MATCH "serial:\s+7777"
```

Comment:
this can now be tests.nested, since release/2.52 has that tool available, can you either omit commit  https://github.com/snapcore/snapd/pull/10622/commits/b3de1dc15ce408698e61d2a71619df9e7697efe4, or revert that commit in this branch?

Reply:
Nice catch! I revert b3de1dc here I think that is slightly easier going forward with subsequent merges.